eng/pipelines/eng-common-sync.yml

```diff
@@ -30,12 +30,25 @@ stages:
         displayName: Sync ${{ parameters.DirectoryToSync }} Directory
 
         steps:
+        - ${{ if eq(variables['Build.Reason'], 'PullRequest') }}:
+          - pwsh: |
+              $patchDir = "$(System.DefaultWorkingDirectory)/patchfiles"
+              $patchfiles = git format-patch HEAD~1...origin/master -- ${{ parameters.DirectoryToSync }} -o $patchDir
+              if ($patchfiles)
```

Comment:
Is this the correct condition? What is the output of the format-patch command? I would expect it to have output for successes and failures. Perhaps we need to use should use the exit code?

Reply:
The output is the list of all the patch files for success. Failure situations will write to console but not output anything

Reply:
>Failure situations will write to console but not output anything

I'm not sure I understand this. it does output things to the console which is what would go into $patchFiles variable correct?

Reply:
From my testing when it fails it write an info about why it fails to console but the `$patchfiles`  variable remains `$null`.